apps/web/src/utils/wallets.ts

```diff
@@ -48,21 +49,31 @@ export const isPKWallet = (wallet: ConnectedWallet): boolean => {
   return wallet.label.toUpperCase() === WALLET_KEYS.PK
 }
 
-export const isSmartContract = async (address: string, provider?: JsonRpcProvider): Promise<boolean> => {
+const getAccountCode = async (address: string, provider?: JsonRpcProvider): Promise<string> => {
   const web3 = provider ?? getWeb3ReadOnly()
 
   if (!web3) {
     throw new Error('Provider not found')
   }
 
-  const code = await web3.getCode(address)
+  return await web3.getCode(address)
+}
 
+export const isSmartContract = async (address: string, provider?: JsonRpcProvider): Promise<boolean> => {
+  const code = await getAccountCode(address, provider)
   return code !== EMPTY_DATA
 }
 
+export const isEIP7702DelegatedAccount = async (address: string, provider?: JsonRpcProvider): Promise<boolean> => {
+  const code = await getAccountCode(address, provider)
+  return code.startsWith(EIP_7702_DELEGATED_ACCOUNT_PREFIX)
+}
+
 export const isSmartContractWallet = memoize(
   async (_chainId: string, address: string): Promise<boolean> => {
-    return isSmartContract(address)
+    const isContract = await isSmartContract(address)
```

Comment:
nit: with this you fire 2 times the same request to get the account code. Not sure if the web3 provider caches this

Reply:
it should be fine. I checked before doing this whether ethersjs caches this. And it does if the calls are withing 250ms which should be the case.

Reply:
Also just checked it to be 100% sure in the browser network tab - I only see one request when I try to sign. So all good I would say.